File: app/src/main/java/com/example/collections_and_maps/models/benchmarks/ResultItem.java
package com.example.collections_and_maps.models.benchmarks;

import com.example.collections_and_maps.R;

public class ResultItem{
    public static final double EMPTY = -1.0;

    public final int headerText;
    public final int methodName;
    public final boolean progressVisible;
    public final double timing;
    public final int nameForHeader;

    public ResultItem(int headerText, int methodName, double timing, boolean progressVisible) {
        this.headerText = headerText;
        this.methodName = methodName;
        this.timing = timing;
        this.progressVisible = progressVisible;
        this.nameForHeader = headerText == R.string.empty ? methodName : headerText;
    }

    public ResultItem copy(ResultItem oldItem, double timing) {
        return new ResultItem(oldItem.headerText, oldItem.methodName, timing, false);
    }

    public boolean isHeader() {
        return headerText == R.string.empty || methodName == R.string.empty;
    }

    public boolean isResult() {
        return timing > EMPTY;
    }

}

File: app/src/main/java/com/example/collections_and_maps/models/benchmarks/ResultItem.kt
package com.example.collections_and_maps.models.benchmarks

import com.example.collections_and_maps.R

data class ResultItem(
    val headerText: Int,
    val methodName: Int,
    val timing: Double,
    val progressVisible: Boolean
) {
    @JvmField
    val nameForHeader: Int = if (headerText == R.string.empty) methodName else headerText

    constructor(oldItem: ResultItem, timing: Double) : this(
        oldItem.headerText,
        oldItem.methodName,
        timing,
        false
    )

    fun isHeader(): Boolean = headerText == R.string.empty || methodName == R.string.empty

    fun isResult(): Boolean = timing > EMPTY

    companion object {
        const val EMPTY = -1.0
    }
}